static/src/javascripts/projects/common/modules/experiments/tests/weekend-reading-promo.js

```diff
@@ -21,24 +29,35 @@ define([
         this.expiry = '2016-09-12';
         this.author = 'Kate Whalen';
         this.description = 'For just one pageview, show users a banner promoting the Weekend Reading email';
-        this.audience = 1;
+        this.audience = 0.5;
         this.audienceOffset = 0;
         this.successMeasure = 'Snap banner promotes the Weekend Reading email and leads to visitors signing up';
         this.audienceCriteria = 'All visitors who have not yet seen this snap banner';
         this.dataLinkNames = 'weekend-reading-snap';
         this.idealOutcome = 'Visitors click on the CTA and sign-up to the Weekend Reading email';
 
         this.canRun = function () {
-            return !config.page.isAdvertisementFeature && config.page.contentType === 'Article' && !hasSeenDigestSnap();
+            return !config.page.isAdvertisementFeature &&
+                config.page.contentType === 'Article' &&
+                // we ran a similar Snap before, so do not show our new CTA to these users
+                !willClashWithAdblockMessage() &&
+                !hasSeenMessage('habit-digest-message-07-16');
         };
 
-        function hasSeenDigestSnap() {
-            // we ran a similar Snap before, so do not show our new CTA to these users
+        function willClashWithAdblockMessage() {
+            if (adblockMsg.showAdblockMsg()) {
+                return !hasSeenMessage('adblock-message-2016-06-15');
+            } else {
+                return false;
+            }
```

Comment:
`adblockMsg.showAdblockMsg` runs various checks to see if the Adblock message banner should run; I'm borrowing the logic to see if Weekend Reading should defer